server/service/osquery_utils/queries.go

```diff
@@ -544,6 +545,43 @@ var usersQuery = DetailQuery{
 	},
 }
 
+var chromeProfilesQuery = DetailQuery{
+	Query: `SELECT email FROM google_chrome_profiles`, // TODO(mna): where not ephemeral?
+
+	/*
+		IngestFunc: func(logger log.Logger, host *fleet.Host, rows []map[string]string) error {
+			mapping := make([]*fleet.HostDeviceMapping, 0, len(rows))
+			for _, row := range rows {
+				mapping = append(mapping, &fleet.HostDeviceMapping{
+					Email:  row["email"],
+					Source: "google_chrome_profiles",
+				})
+			}
+			host.DeviceMapping = mapping
+
+			return nil
+		},
+	*/
+	// TODO(mna): integrate this once Tomas' MDM PR is merged
+	//DirectIngestFunc: directIngestChromeProfiles,
+}
+
+func directIngestChromeProfiles(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string, failed bool) error {
+	if failed {
+		// assume the extension is not there
+		return nil
+	}
+
+	mapping := make([]*fleet.HostDeviceMapping, 0, len(rows))
+	for _, row := range rows {
+		mapping = append(mapping, &fleet.HostDeviceMapping{
+			Email:  row["email"],
+			Source: "google_chrome_profiles",
+		})
+	}
+	return ds.ReplaceHostDeviceMapping(ctx, host.ID, mapping)
```

Comment:
So we will always delete+insert even if the email is already there. I think in a real situation, the emails will rarely change, I wonder if we should check first whether they have changed before replacing them all to prevent the constant load in the db? Reads can be spread out in readers, so it should scale better.

Reply:
Yeah for this I was kinda waiting to see regarding what Zach mentioned of insert/on duplicate vs read then update/insert. I don't think we should use the read replicas in that case though, as we will insert based on the reads so any tiny lag (there's always one of course) could result in a write failure (or in this case, most likely a duplicate because we don't have a unique index). Even just using the writer, if we read to decide what to insert/delete, we probably need to do the read and write in a transaction to prevent any race.

Reply:
Right, well, we can always see how this behaves in the load test env. Which reminds me: please update agent.go to support this.

Gonna add this to the check list.